src/ctap2/mod.rs

```diff
@@ -809,5 +809,440 @@ pub(crate) fn set_or_change_pin_helper(
+                            public_key,
+                            cred_protect,
+                            large_blob_key: large_blob_key.map(|x| x.into_vec()),
+                        };
+                        credential_result.credential_list[current_rp_backup]
+                            .credentials
+                            .push(key);
+                        if we_are_done {
+                            callback.call(Ok(ManageResult::CredManagement(
+                                CredentialManagementResult::CredentialList(credential_result),
+                            )));
+                            return true;
+                        } else {
+                            continue;
+                        }
+                    }
+                    CredManagementCommand::DeleteCredential(_) => {
+                        callback.call(Ok(ManageResult::CredManagement(
+                            CredentialManagementResult::DeleteSucess,
+                        )));
+                        return true;
+                    }
+                    CredManagementCommand::UpdateUserInformation(_) => {
+                        callback.call(Ok(ManageResult::CredManagement(
+                            CredentialManagementResult::UpdateSuccess,
+                        )));
+                        return true;
+                    }
+                };
+            }
+            Err(e) => {
+                handle_errors!(e, status, callback, pin_uv_auth_result, skip_uv, skip_puap);
+            }
+        }
+    }
+    false
+}
+
+pub(crate) fn configure_authenticator(
+    dev: &mut Device,
+    cfg_subcommand: AuthConfigCommand,
+    status: Sender<crate::StatusUpdate>,
+    callback: StateCallback<crate::Result<crate::ManageResult>>,
+    alive: &dyn Fn() -> bool,
+) {
+    let mut authcfg = AuthenticatorConfig::new(cfg_subcommand);
+    let mut skip_uv = false;
+    let authinfo = match dev.get_authenticator_info() {
+        Some(i) => i.clone(),
+        None => {
+            callback.call(Err(HIDError::DeviceNotInitialized.into()));
+            return;
+        }
+    };
+
+    if authinfo.options.authnr_cfg != Some(true) {
+        callback.call(Err(AuthenticatorError::HIDError(
+            HIDError::UnsupportedCommand,
+        )));
+        return;
+    }
+
+    while alive() {
+        let mut pin_uv_auth_result = PinUvAuthResult::NoAuthRequired;
+        // We can use the AuthenticatorConfiguration-command only in two cases:
+        // 1. The device also supports the uv_acfg-permission (otherwise we can't establish a PUAP)
+        // 2. The device is NOT protected by PIN/UV (yet). This allows organizations to configure
+        //    the token, before handing them out.
+        if authinfo.device_is_protected() {
+            // If authinfo.options.uv_acfg is not supported, this will return UnauthorizedPermission
+            pin_uv_auth_result = match determine_puap_if_needed(
```

Comment:
Why do we need to check `device_is_protected()` first? I thought we had `can_skip_user_verification` for this purpose.

Reply:
Yes, it makes sense to move this check there.